apps/api/src/campaign-application/campaign-application.service.ts

```diff
@@ -88,8 +94,99 @@ export class CampaignApplicationService {
+    }
+
+    let editedCampaignApplication
+    if (isAdminFlag === false) {
+      editedCampaignApplication = await this.prisma.campaignApplication.update({
+        where: { id: id },
+        data: {
+          campaignName: updateCampaignApplicationDto?.campaignName,
+          organizerName: updateCampaignApplicationDto?.organizerName,
+          organizerEmail: updateCampaignApplicationDto?.organizerEmail,
+          organizerPhone: updateCampaignApplicationDto?.organizerPhone,
+          beneficiary: updateCampaignApplicationDto?.beneficiary,
+          organizerBeneficiaryRel: updateCampaignApplicationDto?.organizerBeneficiaryRel,
+          goal: updateCampaignApplicationDto?.goal,
+          history: updateCampaignApplicationDto?.history,
+          amount: updateCampaignApplicationDto?.amount,
+          description: updateCampaignApplicationDto?.description,
+          campaignGuarantee: updateCampaignApplicationDto?.campaignGuarantee,
+          otherFinanceSources: updateCampaignApplicationDto?.otherFinanceSources,
+          otherNotes: updateCampaignApplicationDto?.otherNotes,
+          category: updateCampaignApplicationDto?.category,
+        },
+      })
+
+      delete editedCampaignApplication.state
+      delete editedCampaignApplication.ticketURL
+      delete editedCampaignApplication.archived
+    } else {
+      editedCampaignApplication = await this.prisma.campaignApplication.update({
+        where: { id: id },
+        data: {
+          campaignName: updateCampaignApplicationDto?.campaignName,
+          organizerName: updateCampaignApplicationDto?.organizerName,
+          organizerEmail: updateCampaignApplicationDto?.organizerEmail,
+          organizerPhone: updateCampaignApplicationDto?.organizerPhone,
+          beneficiary: updateCampaignApplicationDto?.beneficiary,
+          organizerBeneficiaryRel: updateCampaignApplicationDto?.organizerBeneficiaryRel,
+          goal: updateCampaignApplicationDto?.goal,
+          history: updateCampaignApplicationDto?.history,
+          amount: updateCampaignApplicationDto?.amount,
+          description: updateCampaignApplicationDto?.description,
+          campaignGuarantee: updateCampaignApplicationDto?.campaignGuarantee,
+          otherFinanceSources: updateCampaignApplicationDto?.otherFinanceSources,
+          otherNotes: updateCampaignApplicationDto?.otherNotes,
+          category: updateCampaignApplicationDto?.category,
+          state: updateCampaignApplicationDto?.state,
+          ticketURL: updateCampaignApplicationDto?.ticketURL,
+          archived: updateCampaignApplicationDto?.archived,
+        },
+      })
+    }
+
+    if (files) {
+      const existingCampaignApplicationFiles = await this.prisma.campaignApplicationFile.findMany({
+        where: { campaignApplicationId: id },
+      })
+
+      for (const file of existingCampaignApplicationFiles) {
+        await this.s3.deleteObject(this.bucketName, file.id)
+      }
+
```

Comment:
Why do we delete the previous files?
I'd say the default is that we incrementally add new files.
If the user wants to delete a file - they should do so explicitly one by one.